components/table/src/index.js

```diff
@@ -86,7 +93,7 @@ const Table = ({ name, names = [], rowIncludesHeading, titles, rows, flexibleCol
           {titles.map((title, index) => (
             // disable false-positive rule - this is an access into an array of strings, not object access
             // eslint-disable-next-line security/detect-object-injection
-            <TableHeading key={title.key || index} name={names[index]}>
```

Comment:
i'd prefer this to remain as it was

whilst the use-case for headings being named may be slightly dubious, we shouldn't prevent that from happening